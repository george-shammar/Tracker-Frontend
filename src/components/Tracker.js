import React from 'react';
import PropTypes from 'prop-types';
import '../stylesheets/tracker.css';

const Tracker = ({ tracker }) => (
  <div className="tracker">
    <div className="card">
      <p>Blood Pressure:</p>
      <div className="display">
        <p className="figure">{tracker.blood_pressure}</p>
        <p className="unit-two">mmHg</p>
      </div>
    </div>
    <div className="card">
      <p>Blood Glucose:</p>
      <div className="display">
        <p className="figure">{tracker.blood_glucose}</p>
        <p className="unit-two">mmol/L</p>
      </div>
    </div>
  </div>

);

Tracker.propTypes = {
  tracker: PropTypes.instanceOf(Object).isRequired,
};

export default Tracker;
