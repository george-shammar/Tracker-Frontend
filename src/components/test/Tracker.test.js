import React from 'react';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Tracker from '../Tracker';

const tracker = {
  id: 1,
  blood_pressure: 1,
  blood_glucose: 1,
  user_id: 1,
};

it('renders Tracker without crashing', () => {
  const div = document.createElement('div');

  ReactDom.render(
    <BrowserRouter>
      <Tracker tracker={tracker} key={tracker.id} />
    </BrowserRouter>,
    div,
  );

  ReactDom.unmountComponentAtNode(div);
});
