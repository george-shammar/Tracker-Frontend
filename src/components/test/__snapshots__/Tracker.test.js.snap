// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Tracker component correctly 1`] = `
<div
  className="tracker"
>
  <div
    className="card"
  >
    <i
      className="fas heart fa-heartbeat"
    />
    <div>
      <p
        className="text-unit"
      >
        Blood Pressure:
      </p>
      <div
        className="display"
      >
        <p
          className="figure"
        >
          1
        </p>
        <p
          className="unit-two"
        >
          mmHg
        </p>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <i
      className="fas heart fa-signal"
    />
    <div>
      <p
        className="text-unit"
      >
        Blood Glucose:
      </p>
      <div
        className="display"
      >
        <p
          className="figure"
        >
          1
        </p>
        <p
          className="unit-two"
        >
          mmol/L
        </p>
      </div>
    </div>
  </div>
</div>
`;
